Formula/f/flexiblas-blis.rb

```diff
@@ -0,0 +1,97 @@
+class FlexiblasBlis < Formula
+  desc "BLIS runtime backend for FlexiBLAS"
+  homepage "https://www.mpi-magdeburg.mpg.de/projects/flexiblas"
+  url "https://csc.mpi-magdeburg.mpg.de/mpcsc/software/flexiblas/flexiblas-3.4.4.tar.xz"
+  sha256 "f3b4db7175f00434b1ad1464c0fd004f9b9ddf4ef8d78de5a75382a1f73a75dd"
+  license all_of: [
+    "LGPL-3.0-or-later",
+    "LGPL-2.1-or-later", # libcscutils/
+  ]
+  head "https://gitlab.mpi-magdeburg.mpg.de/software/flexiblas-release.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "blis"
+  depends_on "flexiblas"
+  depends_on "gcc" # for gfortran
+
+  on_macos do
+    depends_on "libomp" => :build
+  end
+
+  fails_with :gcc do
+    version "11"
+    cause "Need to build with same GCC as GFortran for LTO"
+  end
+
+  def install
+    # Remove -flat_namespace usage
+    inreplace "src/fallback_blas/CMakeLists.txt", " -flat_namespace\"", '"'
+
+    # Add HOMEBREW_PREFIX path to default searchpath to allow detecting separate formulae
+    inreplace "CMakeLists.txt",
+              %r{(FLEXIBLAS_DEFAULT_LIB_PATH "\${CMAKE_INSTALL_FULL_LIBDIR}/\${flexiblasname}/)"},
+              "\\1:#{HOMEBREW_PREFIX}/lib/${flexiblasname}\""
+
+    args = %W[
+      -DCMAKE_INSTALL_RPATH=#{rpath}
+      -DSYSCONFDIR=#{etc}
+      -DEXAMPLES=OFF
+      -DTESTS=OFF
+      -DATLAS=OFF
+      -DMklSerial=OFF
+      -DMklOpenMP=OFF
+      -DMklTBB=OFF
+      -DOpenBLASSerial=OFF
+      -DOpenBLASPthread=OFF
+      -DOpenBLASOpenMP=OFF
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
+    system "cmake", "--build", "build"
```

Comment:
Just giving this a try, it does build only the specified target but unfortunately `cmake --install` still expects all of the targets to be present so we have to keep globbing the exact files to install. This does help remove the need for brewed GCC and OpenMP (on macOS), though.

Reply:
> unfortunately `cmake --install` still expects all of the targets to be present so we have to keep globbing the exact files to install.

Yeah. Would need upstream changes to allow that. A quick example would be COMPONENT per install, e.g.
```diff
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 02cf2e81..e0abf10b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -455,7 +455,7 @@ MACRO(BuildBackendBlas VENDOR BACKEND_CODE)
     ENDIF()
 
     TARGET_LINK_LIBRARIES(flexiblas_${ivendor} ${${VENDOR}_LIBRARY})
-    INSTALL(TARGETS flexiblas_${ivendor} LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}/${FLEXIBLAS_LIBRARY_DIR})
+    INSTALL(TARGETS flexiblas_${ivendor} LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}/${FLEXIBLAS_LIBRARY_DIR} COMPONENT flexiblas_${ivendor})
 
     LIST(APPEND EXTRA_BLAS "${VENDOR}")
     SET(${VENDOR}_BLAS_LIBRARIES "${${VENDOR}_LIBRARY}" PARENT_SCOPE)
@@ -468,7 +468,7 @@ MACRO(BuildBackendBlas VENDOR BACKEND_CODE)
     UNSET(LIBNAME CACHE)
     SET(LIBNAME "${CMAKE_SHARED_LIBRARY_PREFIX}flexiblas_${ivendor}${CMAKE_SHARED_LIBRARY_SUFFIX} ")
     FILE(WRITE "${CMAKE_BINARY_DIR}/${FLEXIBLAS_RC_DIR}/${VENDOR}.conf" "[${VENDOR}]\nlibrary = ${LIBNAME}\n\n")
-    INSTALL(FILES "${CMAKE_BINARY_DIR}/${FLEXIBLAS_RC_DIR}/${VENDOR}.conf" DESTINATION "${CMAKE_INSTALL_FULL_SYSCONFDIR}/${FLEXIBLAS_RC_DIR}")
+    INSTALL(FILES "${CMAKE_BINARY_DIR}/${FLEXIBLAS_RC_DIR}/${VENDOR}.conf" DESTINATION "${CMAKE_INSTALL_FULL_SYSCONFDIR}/${FLEXIBLAS_RC_DIR}" COMPONENT flexiblas_${ivendor})
     # Check for default setting
     STRING (TOLOWER  "${FLEXIBLAS_DEFAULT}" FDB_LOW)
     IF ( FDB_LOW STREQUAL ivendor)
```

Which would allow doing `cmake --install build --component <name>`.